src/ucp/rndv/rndv.c

```diff
@@ -1536,21 +1507,30 @@ static ucs_status_t ucp_rndv_progress_am_zcopy_multi(uct_pending_req_t *self)
 UCS_PROFILE_FUNC_VOID(ucp_rndv_send_frag_put_completion, (self),
                       uct_completion_t *self)
 {
-    ucp_request_t *freq = ucs_container_of(self, ucp_request_t,
-                                           send.state.uct_comp);
-    ucp_request_t *req  = freq->super_req;
+    ucp_request_t *freq  = ucs_container_of(self, ucp_request_t,
+                                            send.state.uct_comp);
+    ucp_request_t *fsreq = freq->super_req;
+    ucp_request_t *sreq  = fsreq->super_req;
```

Comment:
do we have 3 levels of requests now?

Reply:
we had it even before my changes: `sreq` was referenced in `ucp_rndv_send_frag_atp()`
now, we pass a `sreq` to `ucp_rndv_complete_rma_put_zcopy()`